sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -429,64 +305,124 @@ def __init__(self, message):
+            timestamp = self.message.annotations.get(_X_OPT_ENQUEUED_TIME)
+            if timestamp:
+                in_seconds = timestamp/1000.0
+                return datetime.datetime.utcfromtimestamp(in_seconds)
+        return None
 
-    def abandon(self):
-        """A PeekMessage cannot be abandoned. Raises `TypeError`."""
-        raise TypeError("Peeked message cannot be abandoned.")
+    @property
+    def scheduled_enqueue_time(self):
+        if self.message.annotations:
+            timestamp = self.message.annotations.get(_X_OPT_SCHEDULED_ENQUEUE_TIME)
+            if timestamp:
+                in_seconds = timestamp/1000.0
+                return datetime.datetime.utcfromtimestamp(in_seconds)
+        return None
 
-    def defer(self):
-        """A PeekMessage cannot be deferred. Raises `TypeError`."""
-        raise TypeError("Peeked message cannot be deferred.")
+    @property
+    def sequence_number(self):
+        if self.message.annotations:
+            return self.message.annotations.get(_X_OPT_SEQUENCE_NUMBER)
+        return None
 
 
-class DeferredMessage(Message):
-    """A message that has been deferred.
+class ReceivedMessage(PeekMessage):
+    """
+    A Service Bus Message received from service side.
 
-    A deferred message can be completed,
-    abandoned, or dead-lettered, however it cannot be deferred again.
+    .. admonition:: Example:
 
+        .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+            :start-after: [START receive_complex_message]
+            :end-before: [END receive_complex_message]
+            :language: python
+            :dedent: 4
+            :caption: Checking the properties on a received message.
     """
-
-    def __init__(self, message, mode):
-        self._settled = mode == 0
-        super(DeferredMessage, self).__init__(None, message=message)
+    def __init__(self, message, mode=ReceiveSettleMode.PeekLock):
+        self._settled = (mode == ReceiveSettleMode.ReceiveAndDelete)
+        super(ReceivedMessage, self).__init__(message=message)
 
     def _is_live(self, action):
-        if not self._receiver:
-            raise ValueError("Orphan message had no open connection.")
-        super(DeferredMessage, self)._is_live(action)
+        # pylint: disable=no-member
+        if not self._receiver or not self._receiver._running:
+            raise MessageSettleFailed(action, "Orphan message had no open connection.")
+        if self.settled:
+            raise MessageAlreadySettled(action)
+        try:
+            if self.expired:
+                raise MessageLockExpired(inner_exception=self.auto_renew_error)
+        except TypeError:
+            pass
+        try:
+            if self._receiver.session and self._receiver.session.expired:  # pylint: disable=protected-access
+                raise SessionLockExpired(inner_exception=self._receiver.session.auto_renew_error)
+        except TypeError: #TODO: Exception: AttributeError?
```

Comment:
I've still not entirely settled myself that we're better off with an always-present session property vs. returning a SessionReceiver with the methods/property present in the case that session_id is specified in creation.  

In either case this TODO should be removed, but it was an apt reminder that it seems a bit wonky.

Reply:
the former is being addressed in time by having a distinct SessionReceiver; an issue has been created to track this.